packages/material-ui-styles/src/makeStyles.js

```diff
@@ -154,21 +148,46 @@ function detach({ state, theme, stylesOptions, stylesCreator }) {
   if (sheetManager.refs === 0) {
     multiKeyStore.delete(stylesOptions.sheetsManager, stylesCreator, theme);
     stylesOptions.jss.removeStyleSheet(sheetManager.staticSheet);
-
     if (sheetsRegistry) {
       sheetsRegistry.remove(sheetManager.staticSheet);
     }
   }
 
   if (state.dynamicSheet) {
     stylesOptions.jss.removeStyleSheet(state.dynamicSheet);
-
     if (sheetsRegistry) {
       sheetsRegistry.remove(state.dynamicSheet);
     }
   }
 }
 
+function useSynchronousEffect(func, values) {
```

Comment:
@oliviertassinari What is this? It's not a hook. Please explain what you want to do here and not just add something that claims to be a hook but is none. What do you mean by synchronous? `useLayoutEffect` is called synchronous so maybe you mean that instead? Or do you mean `useRenderEffect`? A side effect in render is a bug.

Reply:
> It's not a hook.

What's the *definition* of a hook? Could we consider any function that uses a React hook, e.g. `React.useEffect()`, a hook?

> useLayoutEffect()

This hook isn't called server side. We can't use it. We need to collect the styles server side.

> Or do you mean useRenderEffect?

Sure, we can rename the function like this.

> A side effect in render is a bug.

Do you have a better alternative? (Same strategy by [emotion](https://github.com/emotion-js/emotion/blob/1b262bec34479d0899187c3b0f000e872096742f/packages/core/src/jsx.js#L53) and [styled-components](https://github.com/styled-components/styled-components/blob/fd50880f6a60b9510c8d30d155facd38e40d77a5/packages/styled-components/src/models/StyledComponent.js#L122)))

Reply:
> Please explain what you want to do here 

I'm mounting the style sheet synchronously in the render method. I'm unmounting the stylesheet asynchronous after the component unmount. That's the object.